### ISSUE TITLE
Manage Subscriptions

### ISSUE THREAD

**Issue by heathergreerklein** [DSpace/DSpace#2871]
From the release plan spreadsheet

Estimate from release plan: 16 hours
Expressing interest: none
 
Allow users to subscribe to Collections for daily emails. Extra subscription endpoint (subscription resource) which maps to the subscription table.  May be a subresource of user. UNCERTAIN

**Comment by tdonohue** [comment]
Discussion in dspace-community notes that email subscriptions that already exist will still work (as the backend code still exists).  It's only the REST endpoint to manage them that is missing. https://groups.google.com/g/dspace-community/c/uPQg9Ex_Zi4/m/LSHj-E8kAQAJ

**Comment by abollini** [comment]
we want to note that the subscription functionality have been extended and implemented in DSpace-CRIS 7.
If what we have done in DSpace-CRIS is of interest of the DSpace community and can be accepted as is eventually disabling part of the feature that cannot be immediately ported we are available to make a more accurate estimation for the porting work.
We can anticipate that it should took not less than 32 hours on the REST side.

The feature has been improved on this aspects:
- it is possible to subscribe also community and item
- subscriptions have a type: content (default of dspace) or statistics
- frequency of notification can be defined by the user (daily, weekly, monthly)

**Comment by tdonohue** [comment]
@abollini : The proposed new feature enhancements seem reasonable to me, provided that they don't make this PR overly complex to review/test.  

If they make this code contribution overly complex, then I'd recommend seeing if there are places where the ported code can be easily simplified initially (e.g. maybe there's initially only the "content" subscription type, and we can add in statistics later, or similar).  If it's easier to contribute all these features at once, that's OK, but we may need to make sure we have reviewers with enough time to fully review it. 

In other words, if you feel this feature will be very large & more complex to review, we may want to talk about your DSpace-CRIS improvements in more detail in an upcoming meeting, or get a brief demo of this feature in DSpace-CRIS.

In any case, for now, I'll assign to your team for 7.5 & add the estimate.  If you feel it'd be worth talking through more in an upcoming meeting, let me know and we can make time for a demo of the DSpace-CRIS feature.
